www/js/onboarding/WelcomePage.tsx

```diff
@@ -20,12 +20,22 @@ const WelcomePage = () => {
   const [infoPopupVis, setInfoPopupVis] = useState(false);
   const [existingToken, setExistingToken] = useState('');
 
+  const checkURL = function (result) {
+    let notCancelled = result.cancelled == false;
+    let isQR = result.format == "QR_CODE";
+    let hasPrefix = result.text.split(":")[0] == "emission";
+    let hasToken = result.text.includes("login_token?token");
+
+    logDebug("QR code " + result.text + " checks: cancel, format, prefix, params " + notCancelled + isQR + hasPrefix + hasToken);
```

Comment:
@Abby-Wheelis instead of this working, but hacky solution, I think you should use a standard URL parser. 
https://developer.mozilla.org/en-US/docs/Web/API/URL
I have merged this for now, but I think you should fix it in a cleanup change.

Reply:
started a cleanup at #1083, ready to go pending verification using a real phone -- aiming to do that tonight